src/Grenade/Core/Network.hs

```diff
@@ -44,6 +44,7 @@ import           Data.Kind (Type)
 import           Grenade.Core.Layer
 import           Grenade.Core.LearningParameters
 import           Grenade.Core.Shape
+import           Control.DeepSeq
```

Comment:
For packages outside this repo, I really like to make all imports explcit or qualified.

I think you want:
```
import           Control.DeepSeq (NFData (..), deepseq)
```